hudi-spark-datasource/hudi-spark3-common/pom.xml

```diff
@@ -166,7 +166,7 @@
 
         <dependency>
             <groupId>org.apache.spark</groupId>
-            <artifactId>spark-sql_2.12</artifactId>
+            <artifactId>spark-sql_${spark3.scala.binary.version}</artifactId>
```

Comment:
Is this necessary since `spark3.scala.binary.version` should always be `2.12`?

Reply:
I think we can keep this as scala 2.13 is supported as well. https://mvnrepository.com/artifact/org.apache.spark/spark-sql so if we decide in future to upgrade we can make the change at root pom level.